app/services/ccms/parsers/applicant_search_response_parser.rb

```diff
@@ -36,8 +49,34 @@ def extracted_record_count
         text_from(RECORD_COUNT_PATH)
       end
 
-      def extracted_applicant_ccms_reference(index = 0)
-        doc.xpath(APPLICANT_CCMS_REFERENCE_PATH)[index]&.text
+      def client_list
+        @client_list ||= Hash.from_xml(doc.xpath("//Body//ClientInqRS//ClientList").to_s)&.deep_transform_keys { |k| k.underscore.to_sym }
+      end
+
+      def build_client_result_array
+        all_data = multiple_clients_returned? ? client_list[:client_list][:client] : [client_list[:client_list][:client]]
+        all_data.each_with_object([]) do |client_data, payload|
+          payload << ClientResult.new(first_initial: client_data[:name][:first_name].first,
+                                      last_name: client_data[:name][:surname],
+                                      last_name_at_birth: client_data[:name][:surname_at_birth],
+                                      date_of_birth: client_data[:date_of_birth],
+                                      national_insurance_number: client_data[:ni_number],
+                                      client_reference: client_data[:client_reference_number])
+        end
+      end
+
+      def parse_client(client_struct)
+        client_struct.match = [
+          @applicant.first_name.first.casecmp?(client_struct.first_initial),
+          @applicant.last_name.casecmp?(client_struct.last_name),
+          @applicant.surname_at_birth.casecmp?(client_struct.last_name_at_birth),
+          @applicant.date_of_birth.strftime("%Y-%m-%d").eql?(client_struct.date_of_birth),
+          (@applicant.national_insurance_number.nil? ? "" : @applicant.national_insurance_number).casecmp?(client_struct.national_insurance_number),
```

Comment:
possibly a bit briefer/more readable to just have have (@applicant.national_insurance_number || "").casecmp?(client_struct.national_insurance_number)

Reply:
Thanks @kmahern I remember saying I would investigate the new `.presence` method and then forgot!
I think that would make it 
```ruby
(@applicant.national_insurance_number.presence || "").casecmp?(client_struct.national_insurance_number)
```
I'll check tomorrow :+1:

Reply:
Reviewed and yours was even simpler than mine! 🎉 
Updated :+1: